### PR TITLE
Fork diag/code-action tests to ones without editor dep. 2/N

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/71555

This moves diagnostics tests for C# and VB down to the features layer from the editorfeatures one.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @tmat @genlu

**Comment by CyrusNajmabadi** [comment]
@genlu ptal

File: src/Features/CSharpTest/Diagnostics/AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest_NoEditor.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics
+{
+    public abstract partial class AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest_NoEditor : AbstractDiagnosticProviderBasedUserDiagnosticTest_NoEditor
```
Comment (CyrusNajmabadi) [inline src/Features/CSharpTest/Diagnostics/AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest_NoEditor.cs:16]: post pr will share these again.

**Review by tmat — CHANGES_REQUESTED** [review]
Blocking until https://github.com/dotnet/roslyn/pull/71525 is merged

File: src/Features/CSharpTest/UseVarTestExtensions.cs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#if false
```
Comment (sharwell) [inline src/Features/CSharpTest/UseVarTestExtensions.cs:5]: What's going on here?

File: src/Features/VisualBasicTest/Utils.vb [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Runtime.CompilerServices
+Imports System.Text
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Extensions
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests
```
Comment (sharwell) [inline src/Features/VisualBasicTest/Utils.vb:11]: Was this moved from somewhere?

Comment (CyrusNajmabadi) [reply src/Features/VisualBasicTest/Utils.vb:11]: Yes.  Temporary copy as things move down.  Bring tracked as something to remove once it all moves down.

**Comment by CyrusNajmabadi** [comment]
Sorry, these changes seem independent. Could we move forward with these easy changes that are already passing?

File: src/Features/CSharpTest/UseVarTestExtensions.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharpTest/UseVarTestExtensions.cs:5]: Not needed on this form, and can be removed in followup.

**Review by tmat — APPROVED** [review]
Unblocking. Features test projects are now clean.

File: src/EditorFeatures/VisualBasicTest/Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.vbproj [hunk 4]
```diff
@@ -87,5 +87,4 @@
     <ExpectedCompile Remove="ChangeSignature\AddParameterTests.OptionalParameter.Omit.vb" />
     <ExpectedCompile Remove="ChangeSignature\AddParameterTests.OptionalParameter.SymbolKinds.vb" />
   </ItemGroup>
-  <Import Project="..\..\Analyzers\VisualBasic\Tests\VisualBasicAnalyzers.UnitTests.projitems" Label="Shared" />
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.vbproj:90]: this is big.  all the codestyle tests move down to the features layer from the editorfeatures layer.

File: src/Features/VisualBasicTest/Diagnostics/AbstractVisualBasicDiagnosticProviderBasedUserDiagnosticTest_NoEditor.vb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions
+Imports Xunit.Abstractions
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.Diagnostics
+
+    Partial Public MustInherit Class AbstractVisualBasicDiagnosticProviderBasedUserDiagnosticTest_NoEditor
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasicTest/Diagnostics/AbstractVisualBasicDiagnosticProviderBasedUserDiagnosticTest_NoEditor.vb:12]: being tracked to merge back with AbstractVisualBasicDiagnosticProviderBasedUserDiagnosticTest.

File: src/Features/DiagnosticsTestUtilities/CodeActionsLegacy/AbstractCodeActionOrUserDiagnosticTest_NoEditor.cs [hunk 6]
```diff
@@ -157,7 +157,8 @@ private protected virtual IDocumentServiceProvider GetDocumentServiceProvider()
         protected virtual TestComposition GetComposition()
             => FeaturesTestCompositions.Features
                 .AddExcludedPartTypes(typeof(IDiagnosticUpdateSourceRegistrationService))
-                .AddParts(typeof(MockDiagnosticUpdateSourceRegistrationService));
+                .AddParts(typeof(MockDiagnosticUpdateSourceRegistrationService))
+                .AddAssemblies(typeof(DiagnosticService).Assembly);
```
Comment (CyrusNajmabadi) [inline src/Features/DiagnosticsTestUtilities/CodeActionsLegacy/AbstractCodeActionOrUserDiagnosticTest_NoEditor.cs:161]: DiagnosticService is in LSP layer, which isn't in the feature layer composition (but is in the editor composition).  i feel like this just makes sense ot be in this composition.

File: src/Features/CSharpTest/Microsoft.CodeAnalysis.CSharp.Features.UnitTests.csproj [hunk 7]
```diff
@@ -16,4 +16,5 @@
   <ItemGroup>
     <PackageReference Include="Basic.Reference.Assemblies.Net60" Version="$(BasicReferenceAssembliesNet60Version)" />
   </ItemGroup>
+  <Import Project="..\..\Analyzers\CSharp\Tests\CSharpAnalyzers.UnitTests.projitems" Label="Shared" />
```
Comment (CyrusNajmabadi) [inline src/Features/CSharpTest/Microsoft.CodeAnalysis.CSharp.Features.UnitTests.csproj:19]: this is major.  all the C# code style tests *all* run at the feature layer, not hte editor feature layer.

File: Roslyn.sln [hunk 8]
```diff
@@ -1610,7 +1610,6 @@ Global
 		src\Analyzers\VisualBasic\CodeFixes\VisualBasicCodeFixes.projitems*{0141285d-8f6c-42c7-baf3-3c0ccd61c716}*SharedItemsImports = 5
 		src\Workspaces\SharedUtilitiesAndExtensions\Workspace\VisualBasic\VisualBasicWorkspaceExtensions.projitems*{0141285d-8f6c-42c7-baf3-3c0ccd61c716}*SharedItemsImports = 5
 		src\Compilers\CSharp\csc\CscCommandLine.projitems*{0161e25c-918a-4dc8-9648-30fdcc8e31e9}*SharedItemsImports = 5
-		src\Analyzers\VisualBasic\Tests\VisualBasicAnalyzers.UnitTests.projitems*{0be66736-cdaa-4989-88b1-b3f46ebdca4a}*SharedItemsImports = 5
```
Comment (CyrusNajmabadi) [inline Roslyn.sln:1613]: i have no idea what is going on with VS>  it keeps adding/removing these guys.  @jasonmalinowski do you know what might be causing this?

File: src/EditorFeatures/CSharpTest/Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests.csproj [hunk 9]
```diff
@@ -74,5 +74,4 @@
     <Compile Include="..\..\VisualStudio\Core\Def\CodeCleanup\AbstractCodeCleanUpFixer_Helper.cs" Link="Formatting\AbstractCodeCleanUpFixer_Helper.cs" />
   </ItemGroup>
   <Import Project="$(RepositoryEngineeringDir)targets\ILAsm.targets" />
-  <Import Project="..\..\Analyzers\CSharp\Tests\CSharpAnalyzers.UnitTests.projitems" Label="Shared" />
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests.csproj:77]: this is major.  all the c# code style tests are moving to the feature-layer to test.

File: src/Features/DiagnosticsTestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest_NoEditor.cs [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic.UseAutoProperty;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Diagnostics
```
Comment (CyrusNajmabadi) [inline src/Features/DiagnosticsTestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest_NoEditor.cs:32]: post pr will share these again.

File: Roslyn.sln (hunk 8 above)
Comment (jasonmalinowski) [reply Roslyn.sln:1613]: I don't; @lifengl might?

**Comment by tmat** [comment]
> }

You can delete these and use TestOptions defined in compiler test utils instead.

---
Refers to: src/Features/TestUtilities/Utils/Options.cs:16 in 7447b5b. [](commit_id = 7447b5bcdb6cf650aaffad15011dee45a4593732, deletion_comment = False)

**Pull request merged**